xds/src/main/java/io/grpc/xds/XdsLoadStatsStore.java

```diff
@@ -16,128 +16,95 @@
-import io.envoyproxy.envoy.api.v2.endpoint.EndpointLoadMetricStats;
-import io.envoyproxy.envoy.api.v2.endpoint.UpstreamLocalityStats;
-import io.grpc.xds.ClientLoadCounter.ClientLoadSnapshot;
-import io.grpc.xds.ClientLoadCounter.MetricValue;
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
-import java.util.concurrent.atomic.AtomicLong;
-import javax.annotation.concurrent.NotThreadSafe;
+import javax.annotation.Nullable;
 
 /**
- * An {@link XdsLoadStatsStore} instance holds the client side load stats for a cluster.
+ * Interface for client side load stats store. An {@code XdsLoadStatsStore} maintains load stats for
+ * a service cluster (i.e., GSLB service) exposed by traffic director from a gRPC client's
+ * perspective, including dropped calls instructed by traffic director. Load stats for endpoints
+ * (i.e., Google backends) are aggregated in locality granularity (i.e., Google cluster) while the
+ * numbers of dropped calls are aggregated in cluster granularity.
+ *
+ * <p>An {@code XdsLoadStatsStore} lives the same span of lifecycle as {@link XdsLoadBalancer} and
+ * only tracks loads for localities exposed by remote traffic director. A proper usage should be
+ *
+ * <ol>
+ *   <li>Let {@link XdsLoadStatsStore} track the locality newly exposed by traffic director by
+ *       calling {@link #addLocality(XdsLocality)}.
+ *   <li>Use the locality counter returned by {@link #getLocalityCounter(XdsLocality)} to record
+ *       load stats for the corresponding locality.
+ *   <li>Tell {@link XdsLoadStatsStore} to stop tracking the locality no longer exposed by traffic
+ *       director by calling {@link #removeLocality(XdsLocality)}.
+ * </ol>
+ *
+ * <p>No locality information is needed for recording dropped calls since they are aggregated in
+ * cluster granularity.
+ *
+ * <p>Note implementations should only be responsible for keeping track of loads and generating
+ * load reports with load data, any load reporting information should be opaque to {@code
+ * XdsLoadStatsStore} and be set outside.
  */
-@NotThreadSafe
-final class XdsLoadStatsStore implements StatsStore {
-
-  private final ConcurrentMap<XdsLocality, ClientLoadCounter> localityLoadCounters;
-  // Cluster level dropped request counts for each category specified in the DropOverload policy.
-  private final ConcurrentMap<String, AtomicLong> dropCounters;
-
-  XdsLoadStatsStore() {
-    this(new ConcurrentHashMap<XdsLocality, ClientLoadCounter>(),
-        new ConcurrentHashMap<String, AtomicLong>());
-  }
-
-  @VisibleForTesting
-  XdsLoadStatsStore(ConcurrentMap<XdsLocality, ClientLoadCounter> localityLoadCounters,
-      ConcurrentMap<String, AtomicLong> dropCounters) {
-    this.localityLoadCounters = checkNotNull(localityLoadCounters, "localityLoadCounters");
-    this.dropCounters = checkNotNull(dropCounters, "dropCounters");
-  }
+interface XdsLoadStatsStore {
```

Comment:
nit: Rename to `LoadStatesStore` and `LoadStatesStoreImpl` without `xds` prefix, just as `LocalityStore` and `LocalityStoreImpl`?

Reply:
Wouldn't it be better to have it prefixed with `Xds` as this is part of the top-level xDS implementation? But I would like to hear your opinion so that we could have an agreement on naming. If we end up agree on not prepending `Xds` to top-level components of xDS plugin, I would remove that from all the other classes (e.g., `XdsLoadReportClient`, etc) as well.

Reply:
IMO, package-private classes that don't collide with proto generated class names don't need "Xds" in there names. The package name already has "xds", and the "Xds" in the class names adds no value.

"Xds" is necessary for public classes, as they may be imported and referred to only by the class name.

Reply:
Alright, I removed `Xds` prefix for all of load reporting related classes. @dapengzhang0 Let's follow the above convention when cleaning up classes in xDS load balancer in the future.